src/Microsoft.ML.Transforms/MissingValueReplacing.cs

```diff
@@ -933,28 +893,88 @@ private bool SaveAsOnnxCore(OnnxContext ctx, int iinfo, ColInfo info, string src
+        /// <summary>
+        /// Describes how the transformer handles one column pair.
+        /// </summary>
+        public sealed class ColumnInfo
+        {
+            public enum ReplacementMode : byte
             {
-
+                DefaultValue = 0,
+                Mean = 1,
+                Minimum = 2,
+                Maximum = 3,
             }
 
-            public MissingValueReplacingEstimator(IHostEnvironment env, params MissingValueReplacingTransformer.ColumnInfo[] columns)
+            /// <summary>
+            /// Name of the column resulting from the transformation of <see cref="InputColumnName"/>
+            /// </summary>
+            public readonly string Name;
+            /// <summary>
+            /// Name of column to transform.
+            /// </summary>
+            public readonly string InputColumnName;
+            /// <summary>
+            /// If true, per-slot imputation of replacement is performed.
+            /// Otherwise, replacement value is imputed for the entire vector column. This setting is ignored for scalars and variable vectors,
+            /// where imputation is always for the entire column.
+            /// </summary>
+            public readonly bool ImputeBySlot;
+            /// <summary>
+            /// What to replace the missing value with.
+            /// </summary>
+            public readonly ReplacementMode Replacement;
+
+            /// <summary>
+            /// Describes how the transformer handles one column pair.
+            /// </summary>
+            /// <param name="name">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
+            /// <param name="inputColumnName">Name of column to transform. If set to <see langword="null"/>, the value of the <paramref name="name"/> will be used as source.</param>
+            /// <param name="replacementMode">What to replace the missing value with.</param>
+            /// <param name="imputeBySlot">If true, per-slot imputation of replacement is performed.
+            /// Otherwise, replacement value is imputed for the entire vector column. This setting is ignored for scalars and variable vectors,
+            /// where imputation is always for the entire column.</param>
+            public ColumnInfo(string name, string inputColumnName = null, ReplacementMode replacementMode = Defaults.ReplacementMode,
+                bool imputeBySlot = Defaults.ImputeBySlot)
             {
-                Contracts.CheckValue(env, nameof(env));
-                _host = env.Register(nameof(MissingValueReplacingEstimator));
-                _columns = columns;
+                Contracts.CheckNonWhiteSpace(name, nameof(name));
+                Name = name;
+                InputColumnName = inputColumnName ?? name;
+                ImputeBySlot = imputeBySlot;
+                Replacement = replacementMode;
             }
 
+            internal string ReplacementString { get; set; }
```

Comment:
>ReplacementString [](start = 28, length = 17)

Could it be set via ctor?

Reply:
I am setting it through the constructor and adding a test.

Reply:
Actually, I am not sure we want to expose this.  It's not necessarily something we would like to have in the public API. It's quite confusing to have a string as a "replacementvalue" for missing values because they could be of another type. Under the hood we parse the string to get the replacement value. I think it's better to keep it internal for v1.

Reply:
If you set it via ctor, why do we need `{set;}`?

---
In reply to: [253740813](https://github.com/dotnet/machinelearning/pull/2363#discussion_r253740813) [](ancestors = 253740813)

Reply:
That's because the Command line/Entrypoint API has the options to set a replacement string. We use the `ColumnInfo` object as the main arguments object in the transform. Since we want that logic to work in command line, when we translate the `Arguments` to `ColumnInfo` we want to have an option to set it as well. Making it `readonly` would not allow us to support it for command line.

Reply:
Thank you very much @wschin! Looks a lot better now :)